tests/Evaluator/Strategies/test_full_mixed_strategies_evaluator.py

```diff
@@ -1,43 +1,63 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
 import pytest
 
 from tests.functional_tests.strategy_evaluators_tests.abstract_strategy_test import AbstractStrategyTest
 from evaluator.Strategies import FullMixedStrategiesEvaluator
 from trading.trader.modes import DailyTradingMode
 
 
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+
+
 @pytest.fixture()
 def strategy_tester():
     strategy_tester_instance = TestFullMixedStrategiesEvaluator()
-    strategy_tester_instance.init(FullMixedStrategiesEvaluator, DailyTradingMode)
+    strategy_tester_instance.initialize(FullMixedStrategiesEvaluator, DailyTradingMode)
     return strategy_tester_instance
 
 
 class TestFullMixedStrategiesEvaluator(AbstractStrategyTest):
 
     @staticmethod
-    def test_default_run(strategy_tester):
-        strategy_tester.run_test_default_run(-12)
+    async def test_default_run(strategy_tester):
+        await strategy_tester.run_test_default_run(-10)
 
     @staticmethod
-    def test_slow_downtrend(strategy_tester):
-        strategy_tester.run_test_slow_downtrend(-9.5, -10.5, -20.5, 0)
+    async def test_slow_downtrend(strategy_tester):
+        await strategy_tester.run_test_slow_downtrend(-9.7, -9, -21.5, 0)
 
     @staticmethod
-    def test_sharp_downtrend(strategy_tester):
-        strategy_tester.run_test_sharp_downtrend(-14, -21.5)
+    async def test_sharp_downtrend(strategy_tester):
+        await strategy_tester.run_test_sharp_downtrend(-14.5, -21.5)
```

Comment:
Evaluation update is now after all timeframe update at any given time instead of 1tf update => new orders, next tf update (in the same time) => new order or not.
Basically, now order update is after all required tf are updated (if tf to update are 30min AND 1h, order update is after both are updated instead of 30min => order update, 1h => order update.
This is why there are very small changes sometimes in results

Reply:
OK thanks for your explanation.